src/raqm.c

```diff
@@ -452,12 +449,19 @@ raqm_set_text_utf8 (raqm_t         *rq,
   if (!rq->text_utf8)
     return false;
 
+  unicode = calloc (len, sizeof(uint32_t));
+  if (!unicode)
+    goto fail;
```

Comment:
I think we can do without the `goto` here by just returning `false`, we don’t need to free in case `calloc` returned `NULL`.